pkg/loadbalancer/experimental/priorities.go

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package experimental
+
+import "github.com/cilium/cilium/pkg/source"
+
+// NewSourcePriorities returns sources ordered from the most preferred.
```

Comment:
Introducing the possibility to merge additional sources, and reimplementing `source.AllowOverwrite()` based on the priorities slice is going to change the current behavior (https://github.com/cilium/cilium/pull/35917/files#diff-814092b5dc523be3b01a62d684e8daddae7a8fad6203d20b1d7bae22189bc8e5R135-R139), but hopefully it's fine as this behavior seems illogical.

When trying to follow the approach from your snippet, I'm running into
```
# github.com/cilium/cilium/pkg/source
package github.com/cilium/cilium/pkg/source
	imports github.com/cilium/cilium/pkg/hive
	imports github.com/cilium/cilium/pkg/metrics
	imports github.com/cilium/cilium/pkg/source: import cycle not allowed in test
FAIL	github.com/cilium/cilium/pkg/source [setup failed]
```
Could we leave the support for explicitly merging additional sources for now and just go with _assume "lowest priority"_ for such sources as you suggested above?

Reply:
Oh, perhaps I can use `"github.com/cilium/hive"` instead of `github.com/cilium/cilium/pkg/hive`...

Reply:
It seems to me that all callers of `AllowOverwrite()` are not migrated to hive yet. Which means that if we want to modify the default priorities with `MergeSourcesFunc`, these callers may happen to call `AllowOverwrite()` before this modification (or the modification does not happen at all if the modifying hive cells are not run, which I suppose is expected). This is not a huge issue IMHO, but @ovidiutirla brought it to my attention, so I wanted to point it out (FYI: @joamaki).

Reply:
> but hopefully it's fine as this behavior seems illogical.

Actually, it turns out that `TestNodeLifecycle` actually relied on the illogical behaviour, namely on the possibility for `""` to overwrite `"unspec"`. I [propose](https://github.com/cilium/cilium/pull/35810/commits/8c16bfbf511614c8c9a4cb1c003db80b6a7e2d3e) changing the value of `source.Unspec` from `"unspec"` to `""` to avoid this strange state of having two empty values.

Reply:
I've isolated some changes to https://github.com/cilium/cilium/pull/36095, we can continue the discussion there.